### PR TITLE
【Sparse】add new API/OP(csr->csr) of SparseTensor softmax

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

OPs
### Describe

**新增稀疏Tensor激活函数`paddle.incubate.sparse.nn.Softmax`  与 `paddle.incubate.sparse.nn.functional.softmax`**

paddle.incubate.sparse.nn.Softmax
---
```python
import paddle
import numpy as np
from paddle.fluid.framework import _test_eager_guard

paddle.seed(100)

with _test_eager_guard():
    mask = np.random.rand(3, 4) < 0.5
    np_x = np.random.rand(3, 4) * mask
    # [[0.         0.         0.96823406 0.19722934]
    #  [0.94373937 0.         0.02060066 0.71456372]
    #  [0.         0.         0.         0.98275049]]

    csr = paddle.to_tensor(np_x).to_sparse_csr()
    # Tensor(shape=[3, 4], dtype=paddle.float64, place=Place(gpu:0), stop_gradient=True, 
    #        crows=[0, 2, 5, 6], 
    #        cols=[2, 3, 0, 2, 3, 3], 
    #        values=[0.96823406, 0.19722934, 0.94373937, 0.02060066, 0.71456372,
    #                0.98275049])

    m = paddle.incubate.sparse.nn.Softmax()
    out = m(csr)
    # Tensor(shape=[3, 4], dtype=paddle.float64, place=Place(gpu:0), stop_gradient=True, 
    #        crows=[0, 2, 5, 6], 
    #        cols=[2, 3, 0, 2, 3, 3], 
    #        values=[0.68373820, 0.31626180, 0.45610887, 0.18119845, 0.36269269,
    #                1.        ])
```

paddle.incubate.sparse.nn.functional.softmax
---
```python
import paddle
import numpy as np
from paddle.fluid.framework import _test_eager_guard

paddle.seed(100)

with _test_eager_guard():
    mask = np.random.rand(3, 4) < 0.5
    np_x = np.random.rand(3, 4) * mask
    # [[0.         0.         0.96823406 0.19722934]
    #  [0.94373937 0.         0.02060066 0.71456372]
    #  [0.         0.         0.         0.98275049]]

    csr = paddle.to_tensor(np_x).to_sparse_csr()
    # Tensor(shape=[3, 4], dtype=paddle.float64, place=Place(gpu:0), stop_gradient=True, 
    #        crows=[0, 2, 5, 6], 
    #        cols=[2, 3, 0, 2, 3, 3], 
    #        values=[0.96823406, 0.19722934, 0.94373937, 0.02060066, 0.71456372,
    #                0.98275049])

    out = paddle.incubate.sparse.nn.functional.softmax(csr)
    # Tensor(shape=[3, 4], dtype=paddle.float64, place=Place(gpu:0), stop_gradient=True, 
    #        crows=[0, 2, 5, 6], 
    #        cols=[2, 3, 0, 2, 3, 3], 
    #        values=[0.68373820, 0.31626180, 0.45610887, 0.18119845, 0.36269269,
    #                1.        ])
```

### CODE REVIEW

**Review by chenwhql — DISMISSED** [review]

**Review by lanxianghit — DISMISSED** [review]

File: paddle/phi/kernels/sparse/gpu/softmax_kernel.cu [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+  T exp_sum = 0;
+  for (int i = 0; i < kIteration; ++i) {
+    int idx = non_zero_idx + i * warpSize;
+    if (idx >= row_nnz) break;
+
+    auto functor = phi::funcs::CudaExpFunctor<T>();
+    out_values[row_first + idx] =
+        functor(x_values[row_first + idx] - row_max_val);
+    exp_sum += functor(x_values[row_first + idx] - row_max_val);
```
Comment (zkh2016) [inline paddle/phi/kernels/sparse/gpu/softmax_kernel.cu]: T exp = functor(x_values[row_first + idx] - row_max_val);
out_values[row_first + idx] = exp;
exp_sum += exp;

Comment (zhwesky2010) [reply paddle/phi/kernels/sparse/gpu/softmax_kernel.cu]: Done, thx~

**Review by zkh2016 — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

File: python/paddle/incubate/sparse/nn/layer/activation.py [hunk 2]
```diff
@@ -59,3 +59,72 @@ def forward(self, x):
...
         name_str = 'name={}'.format(self._name) if self._name else ''
         return name_str
+
+
+class Softmax(Layer):
+    """
+    sparse softmax activation, x must be SparseCsrTensor or SparseCooTensor.
+
+    Note:
+        Only supported axis=-1 for SparseCsrTensor, which is faster when read data
```
Comment (dingjiaweiww) [inline python/paddle/incubate/sparse/nn/layer/activation.py:69]: only support axis=-1 ?

Comment (zhwesky2010) [reply python/paddle/incubate/sparse/nn/layer/activation.py:69]: supported写错了，应该是support，下个PR改下哈

**Review by dingjiaweiww — APPROVED** [review]
